### ISSUE TITLE
Macro definitions conflict

### ISSUE THREAD

**Issue by colinfang** [osqp/osqp#234]
`constants.h` defines a lot of macros, many of them are common names, e.g

```c
# define RHO (0.1)
# define SIGMA (1E-06)
# define MAX_ITER (4000)
# define EPS_ABS (1E-3)
# define EPS_REL (1E-3)
# define EPS_PRIM_INF (1E-4)
# define EPS_DUAL_INF (1E-4)
# define ALPHA (1.6)
```

My project has other dependences which happen to use those words. As a result I have to do `#undef` them e.g.

```
#include <osqp.h>
#undef ALPHA
#undef RHO
#undef DELTA
#undef SIGMA
```

Can osql add prefix to the macros?

Or maybe provide a header file that contain a list of `#undef` for all macros it uses so the user can do `#include <cleanup.h>` to clean up the side effects.

**Comment by traversaro** [comment]
Also the `MATLAB` and `PYTHON` macros are quite prone to conflicts, and those can be even trickier to detect as they may change the definition of some specific functions that are defined in the headers. If at some point they could be renamed in `OSQP_MATLAB`  and `OSQP_PYTHON` it would be great.

**Comment by bstellato** [comment]
All macros have now been renamed. It will be fixed in 1.0
